engine/storage/snapshot/src/main/java/org/apache/cloudstack/storage/snapshot/SnapshotDataFactoryImpl.java

```diff
@@ -87,6 +87,9 @@ public SnapshotInfo getSnapshot(DataObject obj, DataStore store) {
     @Override
     public SnapshotInfo getSnapshot(long snapshotId, DataStoreRole role) {
         SnapshotVO snapshot = snapshotDao.findById(snapshotId);
+        if (snapshot == null) {
+            return null;
+        }
```

Comment:
@DaanHoogland just check whether the 'null' returned here is being verified or not wherever this method is called, to avoid any NPE.

Reply:
i ran through the 20 call of getsnapshot(). I added some null checks but think they are superfluent. If we go on with this we should refactor the interface to include a specific exception.